tofhir-engine/src/main/scala/io/tofhir/engine/model/FhirMappingJobExecution.scala

```diff
@@ -11,29 +12,27 @@ import java.util.regex.Pattern
 /**
  * Represents the execution of mapping tasks included in a mapping job.
  *
- * @param id                         Unique identifier for the execution
- * @param projectId                  Unique identifier of project to which mapping job belongs
- * @param job                        Fhir mapping job
- * @param mappingTasks               List of mapping tasks to be executed (as a subset of the mapping tasks defined in the job)
- * @param jobGroupIdOrStreamingQuery Keeps Spark job group id for batch jobs and StreamingQuery for streaming jobs
+ * @param id                             Unique identifier for the execution
+ * @param projectId                      Unique identifier of project to which mapping job belongs
+ * @param mappingTasks                   List of mapping tasks to be executed (as a subset of the mapping tasks defined in the job)
+ * @param jobGroupIdOrStreamingQuery     Keeps Spark job group id for batch jobs and StreamingQuery for streaming jobs
+ * @param isStreamingJob                 Whether the execution is streaming or not
+ * @param hasFileSystemSource            Whether the execution source has a file system source or not
+ * @param fileSystemSourceDataFolderPath If execution has a file system source, this is data folder path of it
+ * @param archiveMode                    Archive mode of execution
+ * @param saveErroneousRecords           Whether to save erroneous records or not
  */
-// TODO: The FhirMappingJobExecution model currently includes the entire FhirMappingJob ('job' field), which is unnecessary.
-//  We should remove the 'job' field from the model. Instead, add only the necessary fields to the model.
-case class FhirMappingJobExecution(id: String = UUID.randomUUID().toString,
-                                   projectId: String = "",
-                                   job: FhirMappingJob,
-                                   mappingTasks: Seq[FhirMappingTask] = Seq.empty,
-                                   jobGroupIdOrStreamingQuery: Option[Either[String, collection.mutable.Map[String, StreamingQuery]]] = None
+case class FhirMappingJobExecution(id: String,
+                                   projectId: String,
+                                   jobId: String,
+                                   mappingTasks: Seq[FhirMappingTask],
+                                   jobGroupIdOrStreamingQuery: Option[Either[String, collection.mutable.Map[String, StreamingQuery]]],
+                                   isStreamingJob: Boolean,
+                                   hasFileSystemSource: Boolean,
```

Comment:
I think this property is redundant. We can simply make fileSystemSourceDataFolderPath property optional and check whether it exists or not instead of using hasFileSystemSource.

Reply:
Optional arguments in case class cause the error:
in object FhirMappingJobExecution, multiple overloaded alternatives of method apply define default arguments.
object FhirMappingJobExecution {.
Instead of making it optional, I made it empty string